src/fwup_apply.c

```diff
@@ -416,7 +416,7 @@ int fwup_apply(const char *fw_filename, const char *task_prefix, int output_fd,
     OK_OR_CLEANUP(fatfs_ptr_callback(&fctx, -1, NULL));
 
     // Flush a subarchive that's being built.
-    OK_OR_CLEANUP(subarchive_ptr_callback(&fctx, "", NULL, NULL));
+    OK_OR_CLEANUP(subarchive_ptr_callback(&fctx, NULL, NULL, NULL));
```

Comment:
I'm surprised that this doesn't crash since the second parameter is dereferenced in `subarchive_ptr_callback`. However, I think your change here is correct and that you uncovered a bug in `subarchive_ptr_callback`. The logic in that `subarchive_ptr_callback` needs to be updated to not dereference the second parameter if it's NULL.